force-app/main/default/classes/PMT_RefundService.cls

```diff
@@ -76,6 +149,92 @@ public with sharing class PMT_RefundService {
         return response;
     }
 
+    public void processRefunds() {
+        Boolean shouldBuildRefundRecord = this.refundRecords == null;
+ 
+        for (npe01__OppPayment__c originalPayment : this.originalPayments) {
+            if (shouldBuildRefundRecord) {
+                buildRefundRecord(originalPayment);
+            }
+            updateParentOpportunity(originalPayment);
+            updateOriginalPayment(originalPayment);
+        }
+    }
+
+    public void processDML() {
```

Comment:
currently, I set a savepoint and roll back the entire database when anything goes wrong because currently, we aren't handling any bulk updates.